packages/react/src/components/field/field.tsx

```diff
@@ -1,12 +1,61 @@
-import { c, classy } from '@onfido/castor';
-import React from 'react';
+import { c, classy, FieldProps as BaseProps, FieldState } from '@onfido/castor';
+import { useForm } from '@onfido/castor-react';
+import React, { SyntheticEvent, useEffect, useRef, useState } from 'react';
+import { FieldProvider } from './useField';
 
 /**
  * Intended to wrap `Input` and `Textarea` components, so that `FieldLabel` and
  * `Validation` can wrap nicely following the minimum content allowed.
  */
-export const Field = ({ className, ...restProps }: FieldProps): JSX.Element => (
-  <div {...restProps} className={classy(c('field'), className)} />
-);
+export function Field({
+  className,
+  disabled,
+  onBlur,
+  onChange,
+  onInvalid,
+  ...restProps
+}: FieldProps): JSX.Element {
+  const form = useForm();
+  const ref = useRef<HTMLDivElement>(null);
+  const [field, setField] = useState<FieldState>(initial);
 
-export type FieldProps = JSX.IntrinsicElements['div'];
+  useEffect(() => update(form), [form]);
+
+  function initial(): FieldState {
+    const reset = () => setField(initial);
+    return { ...form, disabled, reset, validity: {} } as FieldState;
+  }
+
+  const update = (state: Partial<FieldState>) =>
+    setField((field) => ({ ...field, ...state }));
+
+  const readInput = (event: SyntheticEvent<HTMLDivElement>) => {
+    const { disabled, validity } = event.target as HTMLInputElement;
+    update({ disabled, validity });
+  };
+
+  return (
+    <FieldProvider value={field}>
+      <div
+        {...restProps}
+        ref={ref}
```

Comment:
Doesn't this disallow to use `ref` on Field component now?

Reply:
It wasn't being `forwardRef`'d before, so no change here.

However this is completely unused, nice thing you asked, we can just remove it.